# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: 0785d6bfebb3
# Full Hash: 0785d6bfebb3405227469a255389dac80b627f91
# Author: Frédéric Wang <fwang@igalia.com>
# Date: 2025-01-17 21:57:54
# Regressor Bug: 1901492
# File Overlap Count: 1
# Description:
#   Bug 1901492 - Introduce RequireTrustedTypesForDirectiveState enum for CSP. r=smaug
#   
#   Extend the hasPolicyWithRequireTrustedTypesForDirective boolean on
#   nsIContentSecurityPolicy to an enum in order to indicate whether there
#   is actually a require-trusted-types-for directive with "enforce"
# ==============================================================================

diff -r 23ca1773d73a -r 0785d6bfebb3 dom/base/Document.cpp
--- a/dom/base/Document.cpp	Fri Jan 17 13:35:39 2025 +0000
+++ b/dom/base/Document.cpp	Fri Jan 17 13:35:39 2025 +0000
@@ -3730,7 +3730,8 @@
 void Document::SetCsp(nsIContentSecurityPolicy* aCSP) {
   mCSP = aCSP;
   mHasPolicyWithRequireTrustedTypesForDirective =
-      aCSP && aCSP->GetHasPolicyWithRequireTrustedTypesForDirective();
+      aCSP && aCSP->GetRequireTrustedTypesForDirectiveState() !=
+                  RequireTrustedTypesForDirectiveState::NONE;
 }
 
 nsIContentSecurityPolicy* Document::GetPreloadCsp() const {
@@ -3861,7 +3862,8 @@
     mHasPolicyWithRequireTrustedTypesForDirective = false;
   } else {
     mHasPolicyWithRequireTrustedTypesForDirective =
-        mCSP->GetHasPolicyWithRequireTrustedTypesForDirective();
+        mCSP->GetRequireTrustedTypesForDirectiveState() !=
+        RequireTrustedTypesForDirectiveState::NONE;
   }
 
   // Always overwrite the requesting context of the CSP so that any new